client/src/components/ProposalTeacher.jsx

```diff
@@ -181,12 +181,18 @@ function ProposalTeacher(props) {
 
   // USE EFFECT /////////////////////////////////////////////
 
+  const [changeLevel,setChangeLevel]= useState(0);
+  
   useEffect(() => {
-    setSelectedDegreeList([]);
-
+    if(changeLevel >= 3 )
+    {
+      setSelectedDegreeList([]);
+    }
+    setChangeLevel((old)=>old+1);
```

Comment:
What is changeLevel? And what is the purpose of setChangeLevel((old)=>old+1)?

Reply:
It's a control to be sure that the setSelectedDeegreList([]) will be execute after 3 mounting of the state level